server/src/main/java/io/littlehorse/common/model/getable/global/acl/PrincipalModel.java

```diff
@@ -0,0 +1,122 @@
+package io.littlehorse.common.model.getable.global.acl;
+
+import com.google.protobuf.Message;
+import io.grpc.Status;
+import io.littlehorse.common.LHSerializable;
+import io.littlehorse.common.exceptions.LHApiException;
+import io.littlehorse.common.model.AbstractGetable;
+import io.littlehorse.common.model.GlobalGetable;
+import io.littlehorse.common.model.getable.ObjectIdModel;
+import io.littlehorse.common.model.getable.objectId.PrincipalIdModel;
+import io.littlehorse.common.proto.ACLAction;
+import io.littlehorse.common.proto.ACLResource;
+import io.littlehorse.common.proto.Principal;
+import io.littlehorse.common.proto.ServerACL;
+import io.littlehorse.common.proto.TagStorageType;
+import io.littlehorse.sdk.common.exception.LHSerdeError;
+import io.littlehorse.sdk.common.proto.PrincipalId;
+import io.littlehorse.server.streams.storeinternals.GetableIndex;
+import io.littlehorse.server.streams.storeinternals.index.IndexedField;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.commons.lang3.tuple.Pair;
+
+@Getter
+@Setter
+public class PrincipalModel extends GlobalGetable<Principal> {
+
+    private String id;
+    private final List<ServerACLModel> acls = new ArrayList<>();
+
+    private TenantModel tenant;
+
+    public PrincipalModel() {}
+
+    public PrincipalModel(final String id, final List<ServerACLModel> acls, final TenantModel tenant) {
+        this.id = id;
+        this.acls.addAll(acls);
+        this.tenant = tenant;
+    }
+
+    public static PrincipalModel anonymous() {
```

Comment:
We need to have the ability to remove the abilities of the `anonymous` Principal. Is that possible here? I don't like the idea of having this method exposed here, because if we mark in the store that `anonymous` has been deleted, I could easily see how a sloppy method call could result in `anonymous` coming back from the dead with full permissions to do everything.